tensorwaves/data/generate.py

```diff
@@ -0,0 +1,109 @@
+"""Tools to facilitate data sample generation."""
+
+import logging
+
+import numpy as np
+
+from progress.bar import Bar
+
+from tensorwaves.interfaces import (
+    Function,
+    Kinematics,
+    PhaseSpaceGenerator,
+    UniformRealNumberGenerator,
+)
+
+
+def _generate_data_bunch(
+    bunch_size: int,
+    phsp_generator: PhaseSpaceGenerator,
+    random_generator: UniformRealNumberGenerator,
+    intensity: Function,
+    kinematics: Kinematics,
+) -> np.ndarray:
+    phsp_sample, weights = phsp_generator.generate(
+        bunch_size, random_generator
+    )
+    dataset = kinematics.convert(phsp_sample)
+    intensities = intensity(dataset)
+    maxvalue = np.max(intensities)
+
+    uniform_randoms = random_generator(bunch_size, max_value=maxvalue)
+
+    phsp_sample = phsp_sample.transpose(1, 0, 2)
+
+    return (phsp_sample[weights * intensities > uniform_randoms], maxvalue)
+
+
+def generate_data(
+    size: int,
+    phsp_generator: PhaseSpaceGenerator,
+    random_generator: UniformRealNumberGenerator,
+    intensity: Function,
+    kinematics: Kinematics,
+) -> np.ndarray:
+    """Create a data sample based on an intensity."""
+    events = np.array([])
+
+    current_max = 0.0
+    bunch_size = 50000
```

Comment:
Maybe give user the option to specify `bunch_size` through function argument?

Reply:
true